core/blockchain.go

```diff
@@ -3084,7 +3109,18 @@ func (bc *BlockChain) PCRC(header *types.Header) (common.Hash, error) {
 		return common.Hash{}, err
 	}
 
-	if PTZ.Hash() != PTR.Hash() || PTR.Hash() != PTP.Hash() || PTP.Hash() != PTZ.Hash() {
+	// PCRC has failed. Rollback through the prior untwisted prime.
+	if PTR.Hash() != PTP.Hash() {
+		ptrParent := bc.GetBlockByHash(PTR.Hash()).Header().ParentHash[types.QuaiNetworkContext]
+		ptrParentBlock := bc.GetBlockByHash(ptrParent)
+		bc.reorg(bc.CurrentBlock(), ptrParentBlock)
+		return common.Hash{}, errors.New("there exists a prime twist")
+	}
+
+	if PTZ.Hash() != PTR.Hash() {
+		ptzParent := bc.GetBlockByHash(PTZ.Hash()).Header().ParentHash[types.QuaiNetworkContext]
+		ptzParentBlock := bc.GetBlockByHash(ptzParent)
+		bc.reorg(bc.CurrentBlock(), ptzParentBlock)
```

Comment:
Rather than duplicate the same logic twice, it would be better to just write it once in a joined conditional:
```
if PTR.Hash() != PTP.Hash() || PTZ.Hash() != PTR.Hash() {
	ptrParent := bc.GetBlockByHash(PTR.Hash()).Header().ParentHash[types.QuaiNetworkContext]
	ptrParentBlock := bc.GetBlockByHash(ptrParent)
	bc.reorg(bc.CurrentBlock(), ptrParentBlock)
	return common.Hash{}, errors.New("there exists a prime twist")
}
```

[DRY programming](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) is life